Fix github action itegration test execution

chris-sanders opened the pull request.

chris-sanders committed:
Fix github action itegration test execution

The github action was using the same configuration as the template
repository which tests both containers and machine varients of the
teampltes. This doesn't apply to to a charm as it will be created with
only one and the action therefore doesn't need to select.

This isn't covered in the integration testing because a github runner is
not setup to verify github action configuration.

Cynerva reviewed (changes requested):
The changes look good, but it looks like the tests on this repo are failing. Can you look into it?

johnsca committed:
Fix integration tests for upstream updates

johnsca committed:
Re-add charmcraft install step to workflow

The top-level of the integration tests don't use pytest-operator and
currently actions-operator doesn't install the charmcraft snap, instead
relying on pytest-operator to do it as a Python package.

johnsca commented:
MicroK8s test failure is from [lp:1929076](https://bugs.launchpad.net/juju/+bug/1929076) but I'm not sure why it's stuck on trying to clean up the (empty) model.

johnsca committed:
Update integration test to include libjuju fix for sidecar charms

johnsca committed:
Fix pytest temp dir to work with confined snap

johnsca committed:
Fix recursive copying of tox dir during test

johnsca commented:
Going to go ahead and merge since the failures are upstream issues anyway.
